SQL/proc_descriptions/procedure_assign_average_speed_to_all_segments_in_area.md

```diff
@@ -0,0 +1,110 @@
+# Assign average speed to all segments in area Decsription
+## Purpose of this file
+The purpose of this file is to ensure that there are as little misunderstandings as could be. That's why it serves to show how I (Vladyslav Zlochevskyi/zlochina) see what this procedure does. Also this way I might figure out what actually could be tested in this procedure.
+
+## Decsription of the procedure
+- Procedure name: procedure_assign_average_speed_to_all_segments_in_area
+- The __name__ of the procedure + __arguments__ of the procedure implicate computing average speed per segment of an area and adding this speed info to corresponding tables.
+- Input params: `target_area_id:smallint`, `target_area_srid:integer`. Return value: _None_.
+- Target of the procedure: gather some info and insert it into table `nodes_ways_speeds`
+- Flow:
+    - Prepare for final insertion to `nodes_ways_speeds`, by executing CTE such as `average_speed`, `target_ways` and `node_segments`:
+        - CTE `average_speed` aggregates all records from `nodes_ways_speeds` table to one record containing average speed and standard deviation filtered by quality either equal to 1 or 2.
+        - CTE `target_ways` executes function `get_ways_in_target_area`, which returns table used by `node_segments` CTE. Note: Did not figure out the role it demonstrates to the procedure.
+        - CTE `node_segments` returns a list of records with columns `from_id`, `to_id`, `geom`. `geom` column is not used afterwards, `from_id` and `to_id` are gathered from `nodes_ways` table, where the latter is joined by some specific filtering. Note: more on QA
+    - Final insertion is made by a list, which is created by Cross Join aka Cartesian product of `node_segments` and `average_speed`, which actually leaves dimensionality to (`count(*) FROM node_segments` X 1), because of one line from `average_speed`. Isertion into `nodes_ways_speeds` with `quality`=5 is executed on the end.
+- __Simplifying it all__, it takes records from `nodes_ways` table, adds some additional data from the same table, then adds additional columns such as `average_speed` & `average_st_dev` and inserts to `nodes_ways_speeds` with `quality`=5.
+
+## Todo list
+This part is about suggestions of what could be tested. On the end of every point you can see _Status_, which shows what status is on every point, if approven corresponding `approved` should be shown.
+- Testing __procedure__:
+    - test that after execution of this procedure, some records were added to `nodes_ways_speeds` with corresponding values (which would be hardcoded to the assertion). Status: `on review`
```

Comment:
I agree, and there should be multiple such tests, testing both segments with matching speeds, without them and the combination of segments with and without matching speeds in one way

Reply:
Im not really sure what you are referring to, when saying `segments with matching speeds`. I tried to figure it out by representing flow of tables [here](https://viewer.diagrams.net/index.html?tags=%7B%7D&highlight=0000ff&edit=_blank&layers=1&nav=1&title=aastas_table_flow.drawio.png#R5VrbcpswEP0aP7YDiJsfUydN06ZNp%2B5M0yePDDLQAGKEfCFfX2EEGASOM7ENDk82iy7o7OFod9EITILNLYGR%2Bx3byB8pkr0ZgeuRosgqUNlPakkyizEeZwaHeDZvVBqm3jPiRolbl56N4kpDirFPvahqtHAYIotWbJAQvK42W2C%2FOmsEHSQYphb0Resfz6ZuZjUVo7R%2FQZ7j5jPLOl9fAPPGfCWxC2283jGBmxGYEIxp9i%2FYTJCfgpfjkvX73HK3eDCCQnpIB%2BM%2B%2Br25I%2FRZ%2BWbc%2BY9wFd2qH%2FgoK%2Bgv%2BYKv4thzwhS9FSIpOOzJI4TsLfCpeTtojJyAzRun428bEwT5MmmSY0fRhj3ZJ5cGPjPI7C%2F007HBtcX6IsIMbA7qMbSv%2BI3As%2B20%2B6e161E0jaCVjrVm3GI2gpehjdLlSOxq4fn%2BBPuYbCcDiwXSLYvZY0rwE9q5YxvjuZT24GtlU6JNK4hy4RrGaYQDREnCmvAOisq9yels8st1yQ2QU9fd4QXgNsjp6BQjlx5jf7jTXuFATXCg4AaH4Ra1rp2%2FOHCeN5deiwkYVyBRNRETuQkT%2FVSYKAImIdOleLaGSSygw17LKP1rJb7H2EVAC%2Fl2WDzPeHg%2FLwzQenK27HxYUjYM4vY4kzNZa6CraaFmus5NLYXwQLq2k0L0V2cOUQWHLAgOZnu8clrpODqyuTpoAvP1c4qB0b0YaFUx0DoXA1PAhELiIDocNTB6pgb5ZJ3uWXlcynmqK13zVBZDsYHtWrLU7LLuXCIGEhRf4q7VhmxPti0ZdC8ImlHduHSpc0EQo6Yt9YahBaBvWiAmWmUMezlC0AJrX4RA714I9JoQqEbnQiCG9QMSAr1vQtCQUOBLk4EWUHsiA3lBaQdJZDtoyi8xoS52cAj9m9Jaw6Zsc49xxMH%2FhyhNeIkZLimuuoZBRpLHtP9HLb%2F8u3vvesMHz66SihvSB2x1AjfFeEkstGfhuVplGeo%2BhMxmBxLkQ%2Bqtqk9y%2FPqamMCl%2FJ%2BVReGa90rfyC9L0kG0fiG7U6oaDpqyO%2BWcqlFQYGiUVg9ktNKSopyJ0WKcMQz3mJfhHjEV6sJfR8Q9Ly30HXhjoO%2BFfCEvhjlU%2F2gX4R%2FQybZ%2FzIh0%2FEb4eNef2GNTFwGaLNfK73It8Mr8ynvVnFA8xhvem7GwoWQFXenrw92PUXae4d0l2K%2F6cqePxSzwvJk2OGDXP3U1KF9wnkh0%2FjkTiHXI8oPEbHtS593xdg87elQYAmJh6BeiSxIWts7T4tpHT7mJzWdNi8E7jS9PX8Bp3leL82jcw4XivbCvihu0qlUH0rXqQFnkcLINWhVLTPw84oyfRuz4XapVmIpiaGevknrpoSY4NKRXTxTSs8vyNG5G5PJMM7j5Dw%3D%3D#%7B%22pageId%22%3A%228H1runfKuQ5gq1I37ers%22%7D), but nothing really came clear